src/add-group-modal/AddGroupModal.tsx

```diff
@@ -108,18 +114,24 @@ const NewGroupForm = (props) => {
   );
 };
 
-const AddGroupModal = () => {
+const AddGroupModal = (props) => {
+  const {
+    patients,
+    isCreate,
+    groupName,
+    cohortUuid,
+    isOpen,
+    handleCancel,
+    onPostSubmit,
+  } = props;
+
   const { setGroup } = useContext(GroupFormWorkflowContext);
   const { t } = useTranslation();
-  const [open, setOpen] = useState(false);
+  const [, setOpen] = useState(false);
```

Comment:
This seems very bad. `isOpen` should be the default or else we should just get rid of this `useState()` altogether. Without maintaining the `open` variable, the calls to `setOpen()` are kind of superfluous.

 ```suggestion
  const [, setOpen] = useState(isOpen);
```

Alternatively, replace all the `setOpen(false)` calls with `handleCancel()`...